### ISSUE TITLE
tx messages not being logged

### ISSUE THREAD

**Issue by ryuash** [forbole/bdjuno#126]
## Bug description

Originally this was an issue: https://github.com/forbole/big-dipper-desmos/issues/54
referencing because it might be similar

Right now block 100145 has 3 tx but it's did not get logged in bdjuno. The tx exists on the chain itself

https://morpheus.desmos.network/blocks/100145

**Comment by rmontagnin** [comment]
@ryuash I think this might be a partial query bug. Let's consider height `112902` (I'll explain later why this one). 

BigDipper shows no transactions inside the details of the block ([here](https://morpheus.desmos.network/blocks/112902)), however there should be three transactions displayed: 

```grapqhql
query Transactions {
  transaction(where:{block:{height:{_eq:112902}}}) {
    hash
    messages
  }
}
```

<details> 

<summary>Output</summary>

```json
{
  "data": {
    "transaction": [
      {
        "hash": "D70E9E06B979DE70EAB9AA4BE866FC5570A1C497F0EA41FAF9C47995F2B1432C",
        "messages": [
          {
            "@type": "/cosmos.staking.v1beta1.MsgDelegate",
            "amount": {
              "denom": "udaric",
              "amount": "884621"
            },
            "delegator_address": "desmos164kcy3s0pqfall3r837n82l24alc6l36ky56ld",
            "validator_address": "desmosvaloper164kcy3s0pqfall3r837n82l24alc6l36gfuw4l"
          }
        ]
      },
      {
        "hash": "5DA8DA20A974298EC183CCE5B7F61C29313722350CBB277CD60600D373135DC3",
        "messages": [
          {
            "@type": "/cosmos.staking.v1beta1.MsgDelegate",
            "amount": {
              "denom": "udaric",
              "amount": "16737855"
            },
            "delegator_address": "desmos18kvwy5hzcu3ss08lcfcnx0eajuecg69ujmkwjr",
            "validator_address": "desmosvaloper18kvwy5hzcu3ss08lcfcnx0eajuecg69uvk76c3"
          }
        ]
      },
      {
        "hash": "89B86CADFEF3DFF634D21402CD42D63F14D730AA5063C99F71614CAADB1B20EF",
        "messages": [
          {
            "@type": "/cosmos.staking.v1beta1.MsgUndelegate",
            "amount": {
              "denom": "udaric",
              "amount": "2000"
            },
            "delegator_address": "desmos12mu43g2qvdddfc3tpgr3pekkrdx23jgv36jlta",
            "validator_address": "desmosvaloper157vyrmqsgue9t2zrvwjau2lsdf8ywjx0756w2q"
          }
        ]
      }
    ]
  }
}
```

</details>

Now, you might be asking: 

> Why are there 3 transactions inside the `transaction` table, but then `block.num_txs` is 6? 

This is not an easy question. We're still exploring why this happens, but what we understood till now is that there are some Desmos-related transactions that, when failing, are not removed from the mempool of the blockchain. This results in them being continuously included in new blocks even when they should not. 

Since the `transaction` table has the `hash` as its primary key, this means that the same transaction with the same hash cannot (because it **should not**) appear in two different blocks. For this reason, currently Desmos blocks show to have `n+3` transactions inside, where `n` is the real number of transactions that should be included in the block.

TL:DR. This appears to be a Desmos bug, that should be solved in the future with a Desmos upgrade.

**Comment by ryuash** [comment]
closing since nothing needs to be fixed in big dipper or bdjuno.
